stats/statsd.go

```diff
@@ -297,9 +312,8 @@ type statsdClient struct {
 }
 
 // ready returns true if the statsd client is ready to be used (not nil).
+//
+// sc.statsdMu.RLock should be held when calling this method.
```

Comment:
didn't understood, why are we removing lock from here and asking caller to hold lock? Isn't that same?

Reply:
Discussed offline, along with calling this function in skip() fn call, we are recording stats as well. we would then need to hold lock again [here](https://github.com/rudderlabs/rudder-go-kit/pull/372/files#diff-d0207bfd221e2e4f4d3605a5eef7b02c571774ad2cbd4a48f523ef024d4a34faR124) before recording new stats. Holding multiple locks multiple time can slow application down so we removed lock from here.